### ISSUE TITLE
Weaning off of the MediaStore

### ISSUE THREAD

**Issue by timusus** [timusus/Shuttle#13]
The MediaStore is the worst thing about Shuttle. It's buggy. Slow to update. It doesn't scan certain paths. Often it doesn't pickup song/album year information. It doesn't tidy up genres when they no longer exist. Worst of all, it makes the FileBrowser almost useless. Shuttle is coupled so tightly to the MediaStore that a song cannot be played unless it is present in the MediaStore.

I've started working on this under a separate library: https://github.com/timusus/MediaProvider but it's barely underway (WIP).

**Comment by playmer** [comment]
Forgive me if I'm misunderstanding, would this (MediaProvider) potentially allow songs of codecs that aren't supported by media store? Such as ALAC?

**Comment by timusus** [comment]
@playmer no, this issue is really just about how files are detected and catalogued in the app. Support for different media types is more to do with file playback. Shuttle currently uses the native `MediaPlayer` for this. 

I don't know a great deal about licensing for various codecs, but attempting to support those not natively supported by Android is not something I'm particularly interested in pursuing - I think it's a lot of work for a problem which can be fairly easily worked around by transcoding the files in question.

**Comment by ildar** [comment]
There are prayers designed to decode on their own, like VLC. I'm not sure
about ALAC though
